ql/src/test/queries/clientpositive/mapjoin_date.q

```diff
@@ -0,0 +1,6 @@
+set hive.auto.convert.join=true;
+
+CREATE TABLE person (fname string, birthDate date);
+INSERT INTO person VALUES ('Victor', '2023-11-27'), ('Alexandre', '2023-11-28');
+
+SELECT * FROM person p1 INNER JOIN person p2 ON p1.birthDate=p2.birthDate;
```

Comment:
Should explain of this query added? Just to confirm that this is really a mapjoin

Reply:
Added explain in
https://github.com/apache/hive/pull/4929/commits/be930513efd36ef6ea08b281abb43025ad2059e6